src/components/common/CurrencyConverter/index.tsx

```diff
@@ -0,0 +1,307 @@
+
+export interface CurrencyConversionProperty {
+  amount: number;
+  currencyCode: string;
+  showSymbol?: boolean;
+}
+
+export type conversionRates = {
+  [key: string]: number;
+};
+
+export type CObject = {
+  convertedAmount: number;
+  currencySymbol: string;
+};
+
+export const currencySymbols: { [key: string]: string } = {
+  AED: "د.إ",
+  AFN: "؋",
+  ALL: "L",
+  AMD: "֏",
+  ANG: "ƒ",
+  AOA: "Kz",
+  ARS: "$",
+  AUD: "A$",
+  AWG: "ƒ",
+  AZN: "₼",
+  BAM: "KM",
+  BBD: "Bds$",
+  BDT: "৳",
+  BGN: "лв",
+  BHD: ".د.ب",
+  BIF: "FBu",
+  BMD: "$",
+  BND: "B$",
+  BOB: "Bs.",
+  BRL: "R$",
+  BSD: "$",
+  BTN: "Nu.",
+  BWP: "P",
+  BYN: "Br",
+  BZD: "BZ$",
+  CAD: "C$",
+  CDF: "FC",
+  CHF: "CHF",
+  CLP: "$",
+  CNY: "¥",
+  COP: "$",
+  CRC: "₡",
+  CUP: "$MN",
+  CVE: "Esc",
+  CZK: "Kč",
+  DJF: "Fdj",
+  DKK: "kr",
+  DOP: "RD$",
+  DZD: "دج",
+  EGP: "£",
+  ERN: "Nkf",
+  ETB: "Br",
+  EUR: "€",
+  FJD: "FJ$",
+  FKP: "£",
+  FOK: "kr",
+  GBP: "£",
+  GEL: "₾",
+  GGP: "£",
+  GHS: "₵",
+  GIP: "£",
+  GMD: "D",
+  GNF: "FG",
+  GTQ: "Q",
+  GYD: "$",
+  HKD: "HK$",
+  HNL: "L",
+  HRK: "kn",
+  HTG: "G",
+  HUF: "Ft",
+  IDR: "Rp",
+  ILS: "₪",
+  IMP: "£",
+  INR: "₹",
+  IQD: "ع.د",
+  IRR: "﷼",
+  ISK: "kr",
+  JEP: "£",
+  JMD: "J$",
+  JOD: "د.ا",
+  JPY: "¥",
+  KES: "KSh",
+  KGS: "лв",
+  KHR: "៛",
+  KID: "$",
+  KMF: "CF",
+  KRW: "₩",
+  KWD: "د.ك",
+  KYD: "$",
+  KZT: "₸",
+  LAK: "₭",
+  LBP: "ل.ل",
+  LKR: "Rs",
+  LRD: "$",
+  LSL: "L",
+  LYD: "ل.د",
+  MAD: "د.م.",
+  MDL: "L",
+  MGA: "Ar",
+  MKD: "ден",
+  MMK: "K",
+  MNT: "₮",
+  MOP: "MOP$",
+  MRU: "UM",
+  MUR: "Rs",
+  MVR: "Rf",
+  MWK: "MK",
+  MXN: "$",
+  MYR: "RM",
+  MZN: "MT",
+  NAD: "$",
+  NGN: "₦",
+  NIO: "C$",
+  NOK: "kr",
+  NPR: "Rs",
+  NZD: "NZ$",
+  OMR: "ر.ع.",
+  PAB: "B/.",
+  PEN: "S/",
+  PGK: "K",
+  PHP: "₱",
+  PKR: "Rs",
+  PLN: "zł",
+  PYG: "Gs",
+  QAR: "ر.ق",
+  RON: "lei",
+  RSD: "din",
+  RUB: "₽",
+  RWF: "FRw",
+  SAR: "﷼",
+  SBD: "SI$",
+  SCR: "SR",
+  SDG: "ج.س.",
+  SEK: "kr",
+  SGD: "S$",
+  SHP: "£",
+  SLE: "Le",
+  SLL: "Le",
+  SOS: "Sh",
+  SRD: "$",
+  SSP: "£",
+  STN: "Db",
+  SYP: "£S",
+  SZL: "E",
+  THB: "฿",
+  TJS: "SM",
+  TMT: "m",
+  TND: "د.ت",
+  TOP: "T$",
+  TRY: "₺",
+  TTD: "TT$",
+  TVD: "$",
+  TWD: "NT$",
+  TZS: "Sh",
+  UAH: "₴",
+  UGX: "USh",
+  USD: "$",
+  UYU: "$U",
+  UZS: "сўм",
+  VES: "Bs.S",
+  VND: "₫",
+  VUV: "VT",
+  WST: "T",
+  XAF: "FCFA",
+  XCD: "$",
+  XOF: "CFA",
+  XPF: "F",
+  YER: "﷼",
+  ZAR: "R",
+  ZMW: "ZK",
+  ZWL: "$",
+};
+
+export async function fetchRates(): Promise<conversionRates | undefined> {
+  try {
+    const response = await fetch(
+      "https://v6.exchangerate-api.com/v6/c58fed89b800f3566d6c89f1/latest/USD",
+      {
+        method: "GET",
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error("Failed to fetch currency rates");
+    }
+    const data = await response.json();
+
```

Comment:
This whole block, remove it, thank you

Logging to the console is generally prohibited in this stage. Thanks again

So, I see potential here but please you remember stage 2 bah, static just static,  dont overdo. Thanks 

Waiting for your kind corrections. I need a video of you testing your component

Add screenshots to your pnpm test and lint test. Thanksssss